Automattic/wp-calypso
# Plugins: Decode html-encoded title when installing or reading a plugin

#### Proposed Changes

* The following components display the plugin title from the backend. In some scenarios, for example, when the plugin is already installed, that title could contain html-encoded characters. This PR will use the `decodeEntities` to display it correctly 

|Remove modal | After|
|-------|------|
|![CleanShot 2023-01-27 at 18 12 34@2x](https://user-images.githubusercontent.com/3519124/215150987-8c5b17bd-16d7-4e25-9cd1-617b5611f247.png)| ![CleanShot 2023-01-27 at 18 14 57@2x](https://user-images.githubusercontent.com/3519124/215151446-726c309e-7fc9-4c40-b419-5f891fd374da.png)  |

#### Testing Instructions



- In a Business or eCommerce site
* Purchase a plugin that contains an ampersand character, for example `Health Check & Troubleshooting`
* In the thank you page, make sure you don't see any html-encoded characters, i.e. you don't see things like `&amp;`
* Navigate to the plugin details of the installed plugin, e.g. `plugins/health-check/:siteId`
* Near the `Deactivate` button on the sidebar, click on the ellipsis button and click on `Remove`
* In the dialogue that appears, make sure you don't see any html-encoded characters, i.e. you don't see things like `&amp;`

#### Performance considerations

In one of the scenarios, this change iterates over a list of sites, then for each of the sites, it iterates over a list of plugins and then it spreads the properties of each plugin to update the name. That implies multiple nested loops, **but** the assumptions are that in the worst-case scenario, the number of sites for a user is in the 10s or less, the number of plugins would be in the 30s or less, and the number of properties of a plugin is around 30, so the number of iterations in the worst-case scenario should not impact the performance noticeably.

#### Pre-merge Checklist



~- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~
~- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~


Fixes #70273
